hca/util/__init__.py

```diff
@@ -1,5 +1,84 @@
 """
-This file contains utility functions for the DCP CLI.
+This module contains utilities for the DCP CLI and API bindings. These
+utility classes and functions are used under the hood by both the DCP
+API bindings library and by its CLI. There is no need to use these
+utilities directly unless you are extending DCP client functionality.
+
+``SwaggerClient`` is a base class for a general purpose Swagger API
+client connection manager. User classes such as ``hca.dss.DSSClient``
+extend it as follows:
+
+  class APIClient(SwaggerClient):
+      def __init__(self, *args, **kwargs):
+          super(APIClient, self).__init__(*args, **kwargs)
+          self.commands += [self.special_cli_command]
+
+      def special_cli_command(self, required_argument, optional_argument=""):
+          return {}
+
+Each user class should have a configuration subtree keyed by its name
+(such as ``APIClient`` above) under the DCP-wide config manager
+(available via ``hca.get_config()``; the static defaults for the
+config manager are stored in ``hca/default_config.json``). Within that
+subtree, the key ``swagger_url`` should point to the HTTPS URL
+containing the Swagger API definition that the client is providing an
+interface for. On first use, this API definition will be downloaded
+and saved into the user config directory (for example,
+/Users/Alice/.config/hca) with a name determined by the base64
```

Comment:
is there an update strategy?

Reply:
There is not yet an update strategy. I imagine an update strategy could involve re-downloading the Swagger API definition when the hca package version has changed with respect to what's recorded in the user-specific config file, or if the cached API definition file's mtime is older than a certain time cutoff. But I'd like to land that in a separate PR. Filed #79 to keep track of this.